protocol/auditlog.go

```diff
@@ -53,10 +62,34 @@ func (h *directoryHistory) updateVerifiedSTR(newVerified *DirSTR) {
 // and then verifies the remaining STRs in msg, and
 // finally updates the snapshots if the checks pass.
 // Audit() is called when an auditor receives new STRs
-// from a directory.
+// from a specific directory.
 func (h *directoryHistory) Audit(msg *Response) error {
-	// TODO: Implement as part of the auditor-server protocol
-	return CheckPassed
+	if err := msg.validate(); err != nil {
+		return err.(ErrorCode)
+	}
+
+	strs := msg.DirectoryResponse.(*STRHistoryRange)
+
+	// Since the str[0] is pinned in the audit log
+	// expect that STR[0].Epoch is at least 1
+	if strs.STR[0].Epoch < 1 {
```

Comment:
You're right, this case is covered in the `str.Epoch == verifiedSTR.Epoch` case in `AuditDirectory()`. What I wanted to avoid is re-inserting `str[0]`, but if the checks pass, it wouldn't matter anyway.